pkg/apis/management.cattle.io/v3/machine_types.go

```diff
@@ -310,24 +328,45 @@ type Condition struct {
 	// Last time the condition transitioned from one status to another.
 	LastTransitionTime string `json:"lastTransitionTime,omitempty"`
 	// The reason for the condition's last transition.
+	// +optional
 	Reason string `json:"reason,omitempty"`
 	// Human-readable message indicating details about last transition
+	// +optional
 	Message string `json:"message,omitempty"`
 }
 
 type NodeDriverSpec struct {
```

Comment:
Consider adding `omitempty` to any json tags where `+optional` is specified (and vice versa).

Reply:
I was a little hesitant to do that as I'm not sure if there are any expectations in UI that these fields be present, even if they're not populated. I'll test it out though and see if there are any obvious errors.

Reply:
I looked more into this and couldn't see any specific errors in the UI, but after thinking about it more this type has been around for ~5 years with the current configuration and I'm not convinced we wouldn't be breaking _something_ by doing this